api/apps/articles/model/retrieve.coffee

```diff
@@ -54,6 +54,11 @@ moment = require 'moment'
       { contributing_authors: { $elemMatch: { id: ObjectId input.all_by_author} } }
     ) if input.all_by_author
 
+    # Convert query for articles by vertical
+    query.$or.push(
+      { vertical: { $elemMatch: { id: ObjectId input.vertical} } }
+    ) if input.vertical
+
```

Comment:
This doesn't need the $or since we're not querying for more than one thing -- should suffice to just say 
```
query.vertical = { $elemMatch: { id: ObjectId input.vertical} }  if input.vertical
```